### DISCUSSION TITLE
CORS issue when accessing REST API - can we add official options to allow CORS?

### DISCUSSION THREAD

**Discussion by virgil9306** [firefly-iii/firefly-iii#4309] [Firefly III]
I am running Firefly III version 5.4.6

**Description**
Firstly, thank you for such a great and extensible tool! 

My issue is that there is a CORS error when accessing API from my webapp (personal/private, just for a Raspberry Pi project).

**Extra info**
When accessing my webapp which fetches data from the REST API of Firefly, there is a CORS error. I had to manually allow the domain in the config PHP file, but this solution will probably disappear when I upgrade Firefly.

It would be easier to have an official way to support this. Like a configuration option of which domains to add, or even open all CORS domains just for testing (e.g. running in Docker for local development environment). Is this feasible or on the roadmap?

**Bonus points**

N/A

**Comment by JC5** [comment]
What did you change to make this work, exactly?

**Comment by virgil9306** [comment]
Hi @JC5 

Thanks for the quick reply!

I added the following to `handle()` in `/var/www/html/app/Http/Middleware/SecureHeaders.php`:

```
$response->header('Access-Control-Allow-Origin', '*');
$response->header('Access-Control-Allow-Headers', '*');
```

Obviously the `*` is less than ideal but this was just for local testing purposes.

Additional Notes:
* The endpoint is behind HTTPS (over VPN), so it complained when I hit it from `localhost`

**Comment by JC5** [comment]
You could add those headers in the configuration of your web server as well, right?

**Comment by virgil9306** [comment]
I see. For the Docker container where would those settings go? I believe I
did that but it didn’t seem to work. Perhaps I did it incorrectly.

2021年1月14日(木) 23:13 James Cole <notifications@github.com>:

**Comment by JC5** [comment]
They would go in the configuration of the reverse proxy.

**Comment by virgil9306** [comment]
> They would go in the configuration of the reverse proxy. 

Hi @JC5
Thanks for the response. I greatly appreciate it. 
Apologies if I am asking something redundant. Is your meaning that 1) the Apache settings inside the Docker image should be changed, or 2) that I should be setting the reverse proxy I have sitting in front of that? (In this case Nginx Proxy Manager which handles my certificates etc., forwarding URL to Docker IP:port)

Reference: the last post on this issue (I will try not setting the app URL this time) https://github.com/firefly-iii/firefly-iii/issues/2319#issuecomment-507596212

**Comment by JC5** [comment]
It should be set in the nginx proxy manager. The image + Firefly III set a minimal set of headers, just enough to enforce basic security in most cases. Other settings, in your case, are best at the reverse proxy. This gives you the leeway to reconfigure the proxy as you see fit and it separates concerns, as it's called.

**Comment by virgil9306** [comment]
Thank you for the clarification.

I tried to allow the headers there and it didn’t work but I see now that
it’s not an issue of Firefly III blocking it but rather my configuration is
just incorrect.

Thanks again for your help and the great app.

2021年1月21日(木) 1:36 James Cole <notifications@github.com>:

**Comment by flightlesstux** [comment]
TLDR; use TRUSTED_PROXIES variable like `TRUSTED_PROXIES=*`
